torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -3107,7 +3107,7 @@ def maybe_positional_arg_names(func):
 class BaseHOPVariable(WrapHigherOrderVariable):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.supports_input_mutation = False
```

Comment:
Technically, we haven't supported it end to end with existing hops that inherit from base hop e.g. foreach_map. It seems also difficult to patch the flag at testing time. Since we have a reasonable error message in the inductor (see the added test in test_foreach.py), I just flip this to True.